vehicle/src/Vehicle/Compile/Print/TypingError.hs

```diff
@@ -251,12 +249,17 @@ typeRestrictionError ctx (TypeRestrictionOrigin freeEnv (ident, p) sort typ) _ca
             <+> "to a supported type"
     }
   where
+    supportedTypesDescription = case sort of
+      Left _ -> "The possible valid types for" <+> quotePretty sort <+> "annotated declarations are:"
+      Right RestrictedRecord -> "All fields of a record annotated with" <+> quotePretty sort <+> "must have the same type and must be either:"
+
     supportedTypes = case sort of
-      RestrictedProperty -> ["Bool", "Vector Bool n", "Tensor Bool ns"]
-      RestrictedParameter Inferable -> [pretty NatType]
-      RestrictedParameter NonInferable -> map pretty [BoolType, IndexType, NatType, RatType]
-      RestrictedDataset -> ["List A    " <+> datasetElementTypes, "Vector A n" <+> datasetElementTypes]
-      RestrictedNetwork -> ["Tensor Rat [a_1, ..., a_n] -> Tensor Rat [b_1, ..., b_n]  (where 'a_i' and 'b_i' are all constants at compile time)"]
+      Left RestrictedProperty -> ["Bool", "Vector Bool n", "Tensor Bool ns"]
+      Left (RestrictedParameter Inferable) -> [pretty NatType]
+      Left (RestrictedParameter NonInferable) -> map pretty [BoolType, IndexType, NatType, RatType]
+      Left RestrictedDataset -> ["List A    " <+> datasetElementTypes, "Vector A n" <+> datasetElementTypes]
+      Left RestrictedNetwork -> ["Tensor Rat [a_1, ..., a_n] -> Tensor Rat [b_1, ..., b_n]  (where 'a_i' and 'b_i' are all constants at compile time)"]
+      Right RestrictedRecord -> ["The element of a tensor (e.g. Real), or", "A tensor (e.g. Tensor Real [...])"]
```

Comment:
We should really generate all of these lists automatically from the instances we have in scope. Can you open an issue on the Issue tracker to note this?